### ISSUE TITLE
Clones of models with long names can disappear

### ISSUE THREAD

**Issue by Jedsters** [FrSkyRC/ETHOS-Feedback-Community#3977]
Tested at 1.5.6

To recreate:

Create a basic model with name 
LONG MODEL NAME
now clone it and you will see it created and the cursor selects it.
Now select a different model
If you now go to select the cloned model you will see that it doesn't appear to exist any more. 

This appears to be because cloned model file names are longer than the original file name so if the original file name is long then the new file name can be too long for Ethos to detect (i.e. more than 15 chars).

Of course the clones do actually still exist, you just can't access them at the field.

Thanks

**Comment by 4qr1hu9n7** [comment]
Shorter name. Than you should see it.

**Comment by Jedsters** [comment]
Indeed, a sub-optimal solution would be to limit model names to 13 chars so that when they are cloned they remain visible.

**Comment by Jedsters** [comment]
Thinking about this further, I propose the following.

Currently if you clone a model called e.g. MODELNAME the clone will have a filename of modelname01.bin (or the next free modelnamexx.bin) but still be called MODELNAME internally. We therefore now have 2 models to select from, both called MODELNAME, except they differ in that one will have RF Off and will have a random rx id assigned to it. 

We also have a problem that if you clone MYLONGMODELNAME it will be given a filename of mylongmodelname01.bin (if free etc.) which Ethos can no longer detect. Unless you immediately change this you loose the clone (until you can get to a PC). 

How about when you clone a model, the name is truncated to 13 chars if necessary and 01 added to the end (if free etc.) however I also suggest that the internal modelname is changed to match the new filename. e.g. cloning MYLONGMODELNAME creates a new file called mylongmodelna01.bin (if free etc.) and the model clone is renamed correspondingly i.e. MYLONGMODELNA01 in this example.  

Of course if a model called MYLONGMODELNA01 already exists then it will already have a filename of mylongmodelna01.bin so MYLONGMODELNA02 will be used with mylongmodelna02.bin  so this should avoid conflicts of both model name and file name and filenames which are too long. 

The only minor concern I see is that if someone has MYLONGMODELNAME and MYLONGMODELNA01 already then the clones of these models end up with the same name format, however if others are like me, then if the first 13 chars of a model name are the same it means they are variations of the same model anyway. 

I suspect there is no perfect solution, but this one seems to tick most boxes to me.

Thanks

**Comment by GiorgioCirelli** [comment]
similar proposal made some time ago #2104

**Comment by bsongis-frsky** [comment]
> Indeed, a sub-optimal solution would be to limit model names to 13 chars so that when they are cloned they remain visible.

Then what will happen if you clone a clone?

**Comment by Jedsters** [comment]
With my proposal above it makes no difference if you are cloning models or clones. 
e.g. suppose you have 2 models called
MYLONGMODELNAME and mylongmodelname.bin
You also have 
MYLONGMODELNA02 and mylongmodelna02.bin

You clone MYLONGMODELNAME and as suggested truncate the name to 13 chars and append the next free nn to the MODEL NAME and modelname.bin (I think this is what currently happens with cloning - you use a nn sequence for file names)
In this case that results in MYLONGMODELNA01 and mylongmodelna01.bin

You now clone the clone i.e. you clone MYLONGMODELNA01. Again you truncate to 13 chars (dropping the 01) and append the next free nn
resulting in MYLONGMODELNA03 and mylongmodelna03.bin - because *02 and *02.bin are already in use.

As mentioned, the main down side to this is that you potentially have multiple models using the same name format, but as above, I'd suggest that if the first 13 chars of a model name are the same then if most people are like me in reality they are actually minor variations of the same model anyway. I guess this would be a matter of documenting this so that users are aware of the consequences of using and cloning models with names longer than 13 chars. 

Alternatively you're into the game of when someone clones a model asking them to give it a name and ensuring uniqueness. I can see this could both be good and bad. Bad from the point of view of I've just cloned a model and need to give it a unique name and I can't recall how many clones I've created so what is my next free number and also I just want a clone, I don't care about the name, just do it for me - it would get in the way of what I'm trying to do. 

Or maybe a hybrid solution, when cloning models Ethos proposes a new unique name using the above format which we can accept with a click or change. e.g. I'm cloning MYLONGMODELNA01 above and Ethos gives a pop up suggesting MYLONGMODELNA03 which I can either change or just ENTER Again, Ethos would ensure uniqueness so if I change it to MYLONGMODELNA02 I'd get a 'name in use' message. Best have a 'cancel' here too in case I want to go back and check existing model names before choosing a new one.

The only other way I see around the problem of clones having almost duplicate model name without limiting to 13 chars would be to allow clones to have more than 15 chars - but I can see that would not be desirable from a programming point of view. 

Maybe the hybrid solution is the best for the user?  

EDITED - Github didn't like me using < ENTER > without spaces!
